torch/export/dynamic_shapes.py

```diff
@@ -38,6 +42,64 @@ def readable(name, min_, max_):
             return f"Dim('{name}', min={min_})"
         return f"Dim('{name}', min={min_}, max={max_})"
 
+    def __add__(cls, other):
+        # e.g., dim + 1
+        assert type(other) is int, f"Expected int, got {type(other)}"
+        return cls._derive(f"({cls.__name__}+{other})", lambda x: x + other)
```

Comment:
OOC, is there a reason we want to immediately string format as we are constructing _Dim expressions? Seems a bit unusual.

Reply:
It is. I need to ensure that the name we give to the derived Dim is unique up to structural equality, hence this trick.

Reply:
I ended up reusing the lambda and the printer for symbolic expressions here in the updated PR, FYI.